signcrypt_seal.go

```diff
@@ -246,6 +236,24 @@ func (sss *signcryptSealStream) init(receiverBoxKeys []BoxPublicKey, receiverSym
 	return nil
 }
 
+func sealEncryptionKeyForReceivers(receiverBoxKeys []BoxPublicKey, receiverSymmetricKeys []ReceiverSymmetricKey, ephemeralKey BoxSecretKey, encryptionKey SymmetricKey) []receiverKeys {
+	order := randomPerm(len(receiverBoxKeys) + len(receiverSymmetricKeys))
+
+	receiverKeysArray := make([]receiverKeys, len(receiverBoxKeys)+len(receiverSymmetricKeys))
+	// Collect all the recipient identifiers, and encrypt the payload key for
+	// all of them.
+	for i, receiverBoxKey := range receiverBoxKeys {
+		index := order[i]
+		receiverKeysArray[index] = receiverEntryForBoxKey(receiverBoxKey, ephemeralKey, encryptionKey, uint64(index))
```

Comment:
And even when it's more than one (shared DH keys?), the number of recipients on a message is also expected to be "small".